### PR TITLE
Add support for nav_msgs/OccupancyGrid

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 New feature

Depends on https://github.com/ignitionrobotics/ign-msgs/pull/138/

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: ros_ign_bridge/src/factories.cpp [hunk 1]
```diff
@@ -335,6 +346,30 @@ get_factory_impl(
       >
     >("sensor_msgs/BatteryState", ign_type_name);
   }
+  if (
+    (ros_type_name == "visualization_msgs/Marker" || ros_type_name == "") &&
+     ign_type_name == "ignition.msgs.Marker")
+  {
+    ROS_ERROR_STREAM("visualization_msgs/Marker");
```
Comment (acschang) [inline ros_ign_bridge/src/factories.cpp]: Should this error stream message be here?

File: ros_ign_bridge/src/factories.cpp [hunk 2]
```diff
@@ -335,6 +346,30 @@ get_factory_impl(
...
+        visualization_msgs::Marker,
+        ignition::msgs::Marker
+      >
+    >("visualization_msgs/Marker", ign_type_name);
+  }
+  if (
+    (ros_type_name == "visualization_msgs/MarkerArray" || ros_type_name == "") &&
+     ign_type_name == "ignition.msgs.Marker_V")
+  {
+    ROS_ERROR_STREAM("visualization_msgs/MarkerArray");
```
Comment (acschang) [inline ros_ign_bridge/src/factories.cpp]: Should this error stream message be here?

**Review by acschang — COMMENTED** [review]
I noticed a couple of odd error messages regarding MarkerArray messages when launching despite that functionality working.  The OccupancyGrid message bridge is not yet functional on this branch (you're probably aware of this):
`[ERROR] [1615316299.701832622]: Failed to create a bridge for topic [/model/<robo_name>/grid] with ROS type [nav_msgs/OccupancyGrid] and Ignition Transport type [ignition.msgs.OccupancyGrid]`.

File: ros_ign_bridge/src/factories.cpp (hunk 2 above)
Comment (mjcarroll) [reply ros_ign_bridge/src/factories.cpp]: Nope, removed.

**Comment by mjcarroll** [comment]
Sorry, the message bridge got busted off a bad rebase as I was originally working on the `noetic` branch.  This should be good now.

File: ros_ign_bridge/test/launch/test_ign_subscriber.launch [hunk 3]
```diff
@@ -27,7 +27,7 @@
               /imu@sensor_msgs/Imu@ignition.msgs.IMU
               /laserscan@sensor_msgs/LaserScan@ignition.msgs.LaserScan
               /magnetic@sensor_msgs/MagneticField@ignition.msgs.Magnetometer
-              /actuators@mav_msgs/Actuators@ignition.msgs.Actuators
```
Comment (chapulina) [inline ros_ign_bridge/test/launch/test_ign_subscriber.launch:31]: We shouldn't remove `actuators`

**Review by chapulina — CHANGES_REQUESTED** [review]
The `melodic` branch is supposed to build from source against Citadel, so we'll need to backport and release https://github.com/ignitionrobotics/ign-msgs/pull/138 first.

**Comment by mjcarroll** [comment]
> The melodic branch is supposed to build from source against Citadel, so we'll need to backport and release ignitionrobotics/ign-msgs#138 first.

Backport PR: https://github.com/ignitionrobotics/ign-msgs/pull/143

File: ros_ign_bridge/test/test_utils.h [hunk 4]
```diff
@@ -1631,6 +1667,47 @@ namespace testing
...
+  /// \brief Compare a message with the populated for testing.
+  /// \param[in] _msg The message to compare.
+  void compareTestMsg(const ignition::msgs::OccupancyGrid &_msg)
+  {
+    compareTestMsg(_msg.header());
+
+    EXPECT_EQ(100, _msg.info().map_load_time().sec());
+    EXPECT_EQ(200, _msg.info().map_load_time().nsec());
+    EXPECT_FLOAT_EQ(0.05, _msg.info().resolution());
+    EXPECT_EQ(10, _msg.info().width());
```
Comment (iche033) [inline ros_ign_bridge/test/test_utils.h]: I get a compile warning here:

```
 warning: comparison of integer expressions of different signedness: ‘const int’ and ‘const unsigned int’ [-Wsign-compare]
```

File: ros_ign_bridge/test/test_utils.h [hunk 5]
```diff
@@ -1631,6 +1667,47 @@ namespace testing
...
+    EXPECT_EQ(20, _msg.info().height());
+    compareTestMsg(_msg.info().origin());
+
+    EXPECT_EQ(20 * 10, _msg.data().size());
+    EXPECT_EQ('\1', _msg.data()[0]);
```
Comment (iche033) [inline ros_ign_bridge/test/test_utils.h:1708]: a compile warning here:

```
warning: comparison of integer expressions of different signedness: ‘const int’ and ‘const long unsigned int’ [-Wsign-compare]
```

**Review by chapulina — APPROVED** [review]

**Pull request merged**